.github/workflows/sparse_model_uploader.yml

```diff
@@ -0,0 +1,409 @@
+        secured: ["true"]
+        entry:
+          - { opensearch_version: 2.11.0 }
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Export Arguments
+        run: | 
+          echo "MODEL_ID=${{ github.event.inputs.model_id }}" >> $GITHUB_ENV
+          echo "MODEL_VERSION=${{ github.event.inputs.model_version }}" >> $GITHUB_ENV
+          echo "TRACING_FORMAT=${{ github.event.inputs.tracing_format }}" >> $GITHUB_ENV
+          echo "MODEL_DESCRIPTION=${{ github.event.inputs.model_description }}" >> $GITHUB_ENV
+      - name: Autotracing ${{ matrix.cluster }} secured=${{ matrix.secured }} version=${{matrix.entry.opensearch_version}}
+        run: "./.ci/run-tests ${{ matrix.cluster }} ${{ matrix.secured }} ${{ matrix.entry.opensearch_version }} sparse_trace"
+      - name: Limit Model Size to 2GB
+        run: |
+          upload_size_in_binary_bytes=$(ls -lR ./upload/ | awk '{ SUM += $5} END {print SUM}')
+          size_limit_in_binary_bytes="2147483648" 
+          echo "Model Artifact Size: $upload_size_in_binary_bytes binary bytes"
+          if [ "$upload_size_in_binary_bytes" -ge "$size_limit_in_binary_bytes" ]
+          then
+            echo "The workflow cannot upload the model artifact that is larger than 2GB."
+            exit 1
+          fi
+      - name: License Verification
+        id: license_verification
+        run: |
+          apache_verified=$(<trace_output/apache_verified.txt)
+          if [[ $apache_verified == "True" ]]
+          then
+            echo "license_line=${{ needs.init-workflow-var.outputs.verified_license_line }}" >> $GITHUB_OUTPUT
+            echo "license_info=Automatically Verified" >> $GITHUB_OUTPUT
+          else
+            echo "license_line=${{ needs.init-workflow-var.outputs.unverified_license_line }}" >> $GITHUB_OUTPUT 
+            echo "license_info=Manually Verified" >> $GITHUB_OUTPUT
+          fi
+      - name: Model Description Info
+        id: model_description_info
+        run: |
+          model_description_info="$(<trace_output/description.txt)"
+          echo "model_description_info=- Model Description: $model_description_info" >> $GITHUB_OUTPUT
+          echo "$model_description_info"
+      - name: Upload Artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: upload
+          path: ./upload/
+          retention-days: 5
+          if-no-files-found: error
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+          aws-region: ${{ secrets.MODEL_UPLOADER_AWS_REGION }}
+          role-to-assume: ${{ secrets.MODEL_UPLOADER_ROLE }}
+          role-session-name: model-auto-tracing
+      - name: Dryrun model uploading
+        id: dryrun_model_uploading
+        run: |
+          dryrun_output=$(aws s3 sync ./upload/ s3://${{ secrets.MODEL_BUCKET }}/${{ needs.init-workflow-var.outputs.sparse_model_folder }} --dryrun \
```

Comment:
Please note we upload the model to different s3 buckets before, since the model is trained by Amazon contributors.
https://artifacts.opensearch.org/models/ml-models/amazon/neural-sparse/opensearch-neural-sparse-encoding-v1/1.0.1/torch_script/neural-sparse_opensearch-neural-sparse-encoding-v1-1.0.1-torch_script.zip
Please modify the logics here to support this different buckets behavior. I think we can add another input parameter to indicate whether it's trained by Amazon

(here is the url for dense: https://artifacts.opensearch.org/models/ml-models/huggingface/sentence-transformers/paraphrase-MiniLM-L3-v2/1.0.1/torch_script/sentence-transformers_paraphrase-MiniLM-L3-v2-1.0.1-torch_script.zip)

Reply:
Can't we extend the current workflow for this sparse use case? If not, why?

Reply:
Because the trace logic are different between dense model and sparse model, and in future we need more parameters for sparse model.

Reply:
which trace logic are you referring to here?

how much more parameter? Let's say for a model we need some other parameters. Will we keep adding different workflows? 

How can we extend this solution? May be a json input as addition parameter if any model requires multiple parameters?